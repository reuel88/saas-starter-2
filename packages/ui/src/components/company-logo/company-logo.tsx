import { FC } from "react";
import { Flame } from "lucide-react";
import { cn } from "@repo/utilities";

export type CompanyLogoProps = {
  className?: string;
};

export const CompanyLogo: FC<CompanyLogoProps> = ({ className, ...rest }) => {
  return (
    <div className={cn("inline-block", className)} {...rest}>
      <span className="sr-only">Blaiz Money</span>
      <Flame className="h-9 w-9 text-hue-500" />
    </div>
  );
};
