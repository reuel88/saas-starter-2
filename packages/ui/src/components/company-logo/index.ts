export { CompanyLogo } from "./company-logo";
