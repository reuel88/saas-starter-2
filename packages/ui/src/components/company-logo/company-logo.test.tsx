import { render, screen } from "@testing-library/react";
import { CompanyLogo } from "./company-logo";

describe("CompanyLogo", () => {
  it("should render the company logo", () => {
    // Arrange

    // Act
    const { baseElement } = render(<CompanyLogo data-testid="company-logo" />);
    const logoContainer = screen.getByTestId("company-logo");

    // Assert
    expect(logoContainer).toBeInTheDocument();
    expect(baseElement).toMatchSnapshot();
  });
});
