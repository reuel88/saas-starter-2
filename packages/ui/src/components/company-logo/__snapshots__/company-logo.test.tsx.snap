// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CompanyLogo > should render the company logo 1`] = `
<body>
  <div>
    <div
      class="inline-block"
      data-testid="company-logo"
    >
      <span
        class="sr-only"
      >
        Blaiz Money
      </span>
      <svg
        class="lucide lucide-flame h-9 w-9 text-hue-500"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.5 14.5A2.5 2.5 0 0 0 11 12c0-1.38-.5-2-1-3-1.072-2.143-.224-4.054 2-6 .5 2.5 2 4.9 4 6.5 2 1.6 3 3.5 3 5.5a7 7 0 1 1-14 0c0-1.153.433-2.294 1-3a2.5 2.5 0 0 0 2.5 2.5z"
        />
      </svg>
    </div>
  </div>
</body>
`;
